services/static-webserver/client/source/class/osparc/ui/list/CollaboratorListItem.js

```diff
@@ -149,62 +165,79 @@ qx.Class.define("osparc.ui.list.CollaboratorListItem", {
       });
 
       const accessRights = this.getAccessRights();
+      let currentRole = this.self().ROLES[1];
+      if (this.self().canDelete(accessRights)) {
+        currentRole = this.self().ROLES[3];
+      } else if (this.self().canWrite(accessRights)) {
+        currentRole = this.self().ROLES[2];
+      }
 
-      const makeOwnerButton = new qx.ui.menu.Button(this.tr("Make Owner"));
-      makeOwnerButton.addListener("execute", () => {
-        this.fireDataEvent("makeOwner", {
-          gid: this.getKey(),
-          name: this.getTitle()
-        });
-      });
-      const makeCollabButton = new qx.ui.menu.Button(this.tr("Make Collaborator"));
-      makeCollabButton.addListener("execute", () => {
-        this.fireDataEvent("makeCollaborator", {
-          gid: this.getKey(),
-          name: this.getTitle()
-        });
-      });
-      const makeViewerButton = new qx.ui.menu.Button(this.tr("Make Viewer"));
-      makeViewerButton.addListener("execute", () => {
-        this.fireDataEvent("makeViewer", {
-          gid: this.getKey(),
-          name: this.getTitle()
-        });
-      });
+      // promote/demote actions
+      switch (currentRole.id) {
+        case "read": {
+          const promoteButton = new qx.ui.menu.Button(this.tr("Promote to ") + this.self().ROLES[2].label);
+          promoteButton.addListener("execute", () => {
+            this.fireDataEvent("promoteToCollaborator", {
+              gid: this.getKey(),
+              name: this.getTitle()
+            });
+          });
+          menu.add(promoteButton);
+          break;
+        }
+        case "write": {
+          const promoteButton = new qx.ui.menu.Button(this.tr("Promote to ") + this.self().ROLES[3].label);
+          promoteButton.addListener("execute", () => {
+            this.fireDataEvent("promoteToOwner", {
+              gid: this.getKey(),
+              name: this.getTitle()
+            });
+          });
+          menu.add(promoteButton);
+          const demoteButton = new qx.ui.menu.Button(this.tr("Demote to ") + this.self().ROLES[1].label);
```

Comment:
And something like ":arrow_down: Demote to"

Reply:
Will check 👎